src/amber/cli/commands/generate.cr

```diff
@@ -23,11 +23,22 @@ module Amber::CLI
           template = Template.new("error", ".")
         else
           ensure_name_argument!
-          template = Template.new(args.name, ".", args.fields)
+          # if the app was generated from a recipe and the component can be
+          # generated for the recipe then use the recipe generator
+          if recipe && recipe != "none" && Amber::Recipes::Recipe.can_generate?(args.type, recipe)
```

Comment:
At first glance, having a recipe called "none" seems like it could be represented more clearly with just `nil`. Doing so would clean up the logic here a little as well.